python/lsst/pipe/base/pipelineTask.py

```diff
@@ -46,6 +46,65 @@ def __init__(self, key, numDataIds):
                           "received {} DataIds").format(key, numDataIds))
 
 
+class DatasetTypeDescriptor:
+    """Describe DatasetType and its option for PipelineTask.
+
+    This class contains DatasetType and all relevant options that are used by
+    PipelineTask. Typically this is derived from configuration classes but
+    sub-classes of PipelineTask can also define additional DatasetTypes that
+    are not part of the task configuration.
+
+    Parameters
+    ----------
+    datasetType : `DatasetType`
+    scalar : `bool`, optional
+        `True` if this is a scalar dataset, `None` for dataset types that
+        do not need scalar option.
+    """
+
+    __slots__ = ("_datasetType", "_scalar")
+
+    def __init__(self, datasetType, scalar=None):
+        self._datasetType = datasetType
+        self._scalar = scalar
+
+    @classmethod
+    def fromConfig(cls, datasetConfig):
+        """Make DatasetTypeDescriptor instance from configuration object.
+
+        Parameters
+        ----------
+        datasetConfig :
```

Comment:
Not sure what is going on but sphinx refuses to generate documentation for any of the class attributes/methods (though it makes docs for class and constructor). I'm sure missing type is not a problem here but I will add `object` as a type.

Reply:
Or `lsst.pex.config.Config`

Reply:
I think it's because of `__slots__`, I'll remove `__slots__`, they are not super-useful here. Sphinx does OK without type, still I'll leave `object` as a type.